packages/@expo/cli/src/start/server/metro/withMetroSupervisingTransformWorker.ts

```diff
@@ -0,0 +1,49 @@
+import type { ConfigT as MetroConfig } from '@expo/metro/metro-config';
+import { unstable_transformerPath, internal_supervisingTransformerPath } from '@expo/metro-config';
+
+const debug = require('debug')(
+  'expo:metro:withMetroSupervisingTransformWorker'
+) as typeof console.log;
+
+declare module '@expo/metro/metro-transform-worker' {
+  export interface JsTransformerConfig {
+    expo_customTransformerPath?: string;
+  }
+}
+
+const defaultBabelTransformerPaths = [
+  require.resolve('@expo/metro-config/babel-transformer'),
+  require.resolve('@expo/metro-config/build/babel-transformer'),
+];
+
+export function withMetroSupervisingTransformWorker(config: MetroConfig): MetroConfig {
+  const originalTransformerPath = config.transformerPath;
+  const originalBabelTransformerPath = config.transformer?.babelTransformerPath;
+
+  const hasOriginalTransformPath = originalTransformerPath === unstable_transformerPath;
+  const hasOriginalBabelTransformerPath =
+    !originalBabelTransformerPath ||
+    defaultBabelTransformerPaths.includes(originalBabelTransformerPath);
+  if (hasOriginalTransformPath && hasOriginalBabelTransformerPath) {
+    return config;
+  }
+
+  if (!hasOriginalBabelTransformerPath && hasOriginalTransformPath) {
```

Comment:
I might have missed something, but should this one check only `!hasOriginalTransformPath`

Reply:
oops